Book Us stories

agnes-guillet21 opened the pull request.

agnes-guillet21 committed:
Back :
Test : bookdao : fix bug with save methode

agnes-guillet21 committed:
Back :

BookRestController/BookService : update methode name

agnes-guillet21 committed:
Back :
Test :
add new class test
add new test

agnes-guillet21 committed:
Back :
 connection with Postgres Database

agnes-guillet21 committed:
Front :
add css

William-Schiller commented on back/src/main/java/fr/library/back/book/BookRestController.java:
```diff
@@ -113,7 +113,7 @@ public ResponseEntity<BookDto> getById(@PathVariable Integer id){
      */
     public ResponseEntity <BookDto> getByTitle(String title){
```
Tu n'utilises pas cette entrée du controller (pas de Mapping -> @GetMapping)

William-Schiller commented on back/src/main/java/fr/library/back/book/BookRestController.java:
```diff
@@ -113,7 +113,7 @@ public ResponseEntity<BookDto> getById(@PathVariable Integer id){
         BookDto bookDto;
-        bookDto = bookService.getBookByTitle(title);
+        bookDto = bookService.findBookByTitle(title);
         return new ResponseEntity<>(HttpStatus.OK);
```
Tu ne renvoi pas ton livre

William-Schiller commented on back/src/main/java/fr/library/back/book/BookRestController.java:
```diff
@@ -123,7 +123,7 @@ public ResponseEntity <BookDto> getByTitle(String title){
     public ResponseEntity <BookDto> getByIsbn(@PathVariable Integer isbn){
         BookDto bookDto;
-        bookDto = bookService.getBookByIsbn(isbn);
+        bookDto = bookService.findBookByIsbn(isbn);
```
comme en haut tu ne renvoie pas ton livre, par ailleur tu pourrais te passer d'une ligne pour déclarer ta variable etant donnée que tu lui attribut une valeur tt de sutie après : 
BookDto bookDto = bookService.findBookByIsbn(isbn);

William-Schiller commented on back/src/test/java/fr/library/back/book/BookDaoTest.java:
```diff
@@ -2,36 +2,27 @@
-    }
-
     @Test
     public void testAddBook(){
```
dans le test tu donnes un id au livre mais la méthode save est sensé générer l'element et lui attribuer un id tu pourrait tester qu'en ne donnant pas d'id, l'objet renvoyer par e save ait un ID

agnes-guillet21 committed:
Back  :
WIP :  add relation between image entity and book entity

agnes-guillet21 committed:
Back  :
Author : create entity
Image : create entity, service and DAO

agnes-guillet21 committed:
Front :
book component
add css
add image input

agnes-guillet21 committed:
Front :
add navbar in header + css

agnes-guillet21 committed:
Back:
Image :  delete unnecessary column in image table / object

agnes-guillet21 committed:
Back:
Book : image processing added

agnes-guillet21 committed:
Back:
Author : WIP need to add  processing for add book with author. Need to create author dto and mapper

agnes-guillet21 committed:
Front:
book: add field in formbuilder, create json object,
add attribut in constructor, delete console.log
